pkg/envoy/sds/response.go

```diff
@@ -155,14 +155,14 @@ func (s *sdsImpl) getRootCert(cert certificate.Certificater, sdscert envoy.SDSCe
 
 // Given a requested SDS Cert, this function returns the Service Identities, which match that SDS Cert
 // Example: given "service-cert:namespace/service-account", this will return ServiceIdentity("namespace.service-account.cluster.local")
-func getServiceIdentitiesFromCert(sdscert envoy.SDSCert, serviceIdentity identity.ServiceIdentity, meshCatalog catalog.MeshCataloger) ([]identity.ServiceIdentity, error) {
+func getServiceIdentitiesFromCert(sdscert secrets.SDSCert, serviceIdentity identity.ServiceIdentity, meshCatalog catalog.MeshCataloger) ([]identity.ServiceIdentity, error) {
 	// Program SAN matching based on SMI TrafficTarget policies
 	switch sdscert.CertType {
-	case envoy.RootCertTypeForMTLSOutbound:
+	case secrets.RootCertTypeForMTLSOutbound:
 		// For the outbound certificate validation context, the SANs needs to match the list of service identities
 		// corresponding to the upstream service. This means, if the sdscert.Name points to service 'X',
 		// the SANs for this certificate should correspond to the service identities of 'X'.
-		meshSvc, err := service.UnmarshalMeshService(sdscert.Name)
+		meshSvc, err := sdscert.GetMeshService()
```

Comment:
nit (feel free to ignore):
```suggestion
		meshSvc, err := sdscert.MeshService()
```